Start adding types and cleaning up classes.py

dunngitter opened the pull request:
Like with my last PR I started out just wanting to add some types, but the scope very quickly exploded. This is going to be a huge change and will need pretty thorough testing.

Changes:
- added a `get_logger` method to log.py which can be used instead of `import stashapi.log as log`
- many others I will add here later

Totally untested so far

Todo:
  - [ ] Add tests for everything that has changed
  - [ ] Go over all of the docstrings again and make sure they are consistent
  - [ ] Make the call signatures and behaviors more consistent across the find/create/update/destroy methods for each type

dunngitter committed:
chore: clean-up and add types to classes.py

also start cleaning up stashapp.py and stashbox.py a little bit
added a `get_logger` method to log.py which can be used instead of
`import stashapi.log as log`

dunngitter commented on src/stashapi/stashapp.py:
```diff
@@ -1,32 +1,27 @@
-            import stashapi.log as logger
-
-            self.log = logger
+        super().__init__(conn.get("Logger", None) or get_logger())
```
Is there ever actually a logger in the connection parameters? If so, what does it look like? The documentation on the plugin task inputs doesn't mention this field:

> Plugin task input
Plugin tasks may accept an input from the stash server. This input is encoded according to the interface, and has the following structure (presented here in JSON format):

>{
    "server_connection": {
        "Scheme": "http",
        "Port": 9999,
        "SessionCookie": {
            "Name":"session",
            "Value":"cookie-value",
            "Path":"",
            "Domain":"",
            "Expires":"0001-01-01T00:00:00Z",
            "RawExpires":"",
            "MaxAge":0,
            "Secure":false,
            "HttpOnly":false,
            "SameSite":0,
            "Raw":"",
            "Unparsed":null
        },
        "Dir": <path to stash config directory>,
        "PluginDir": <path to plugin config directory>,
    },
    "args": {
        "argKey": "argValue"
    }
}
The server_connection field contains all the information needed for a plugin to access the parent stash server, if necessary.

stg-annon replied:
logger is a non standard input that is indented to be passed from a plugin so that the log config consistent

dunngitter replied:
Ok cool that was my guess, I think this should work then

dunngitter committed:
chore: add types to tools.py

there were a couple of functions here that I really didn't know what to
do with because I don't see any example uses in the codebase.

dunngitter committed:
chore: stash_types.py use replace StashEnum with StrEnum and IntEnum

dunngitter committed:
chore: more work on types in stashapp.py and classes.py

dunngitter committed:
chore: wip stashapp.py, fixup later

dunngitter committed:
more wip stashapp.py, fixup later
